unitTests/ui/DidYouMeanTest.ts

```diff
@@ -65,61 +65,89 @@ export function DidYouMeanTest() {
+            });
+
+            expect(analyticsSpy).toHaveBeenCalledWith(analyticsActionCauseList.didyoumeanAutomatic, {});
+            expect(analyticsSpy.calls.count()).toBe(1);
           });
 
-          expect(analyticsSpy).toHaveBeenCalledWith(analyticsActionCauseList.didyoumeanAutomatic, {});
-          expect(analyticsSpy.calls.count()).toBe(1);
+          it('should replace the state in history manager when query is autocorrected', () => {
+            spyOn(test.env.searchInterface.historyManager, 'replaceState');
+
+            Simulate.query(test.env, {
+              results: FakeResults.createFakeResults(0),
+              queryCorrections: [fakeQueryCorrection]
+            });
+
+            expect(test.env.searchInterface.historyManager.replaceState).toHaveBeenCalled();
+          });
         });
 
-        it('set to false should not autocorrect the query when no results are found', function() {
-          test = Mock.optionsComponentSetup<DidYouMean, IDidYouMeanOptions>(DidYouMean, <IDidYouMeanOptions>{
-            enableAutoCorrection: false
+        describe('set to false', () => {
+          beforeEach(() => {
+            test = Mock.optionsComponentSetup<DidYouMean, IDidYouMeanOptions>(DidYouMean, <IDidYouMeanOptions>{
+              enableAutoCorrection: false
+            });
           });
-          test.env.queryStateModel.get = () => 'foobar';
-          var qsmSpy = jasmine.createSpy('queryStateModelSpy');
-          test.env.queryStateModel.set = qsmSpy;
 
-          Simulate.query(test.env, {
-            results: FakeResults.createFakeResults(0),
-            queryCorrections: [fakeQueryCorrection]
+          it('should not autocorrect the query when no results are found', () => {
+            test.env.queryStateModel.get = () => 'foobar';
+            const qsmSpy = jasmine.createSpy('queryStateModelSpy');
+            test.env.queryStateModel.set = qsmSpy;
+
+            Simulate.query(test.env, {
+              results: FakeResults.createFakeResults(0),
+              queryCorrections: [fakeQueryCorrection]
+            });
+
+            expect(qsmSpy).not.toHaveBeenCalled();
           });
 
-          expect(qsmSpy).not.toHaveBeenCalled();
+          it('should not replace the state in history manager', () => {
+            spyOn(test.env.searchInterface.historyManager, 'replaceState');
+
+            Simulate.query(test.env, {
+              results: FakeResults.createFakeResults(0),
+              queryCorrections: [fakeQueryCorrection]
+            });
+
+            expect(test.env.searchInterface.historyManager.replaceState).not.toHaveBeenCalled();
+          });
         });
       });
     });
 
-    it('should not autocorrect search-as-you-type queries', function() {
-      var qsmSpy = jasmine.createSpy('queryStateModelSpy');
+    it('should not autocorrect search-as-you-type queries', () => {
+      const qsmSpy = jasmine.createSpy('queryStateModelSpy');
```

Comment:
Same as above, my preference would be to have better readability than to abbreviate